src/states/adult/mod.rs

```diff
@@ -253,7 +253,15 @@ impl Adult {
             .map(|p2p_node| p2p_node.connection_info().clone())
             .collect();
 
-        self.network_service_mut().remove_and_disconnect_all();
+        // Disconnect from everyone we know.
+        let addrs: Vec<_> = self
+            .chain
+            .known_nodes()
+            .map(|node| node.connection_info().peer_addr)
+            .collect();
+        for addr in addrs {
```

Comment:
That seem unfortunate as it does not disconnect from those we do not know.
Is that suggesting there is a missing API here from quick_p2p that we need if we want to not keep track of those we are connected?

Reply:
Do we need to disconnect from those we don't know? But if so, then yeah, we would need support from quic-p2p. Either a method to disconnect from all, or a method to list all connections.

Reply:
I think so, that was what was happening before I think? Yes, I agree.

Reply:
We don't disconnect from unknown peers as elders either, so maybe this issue needs a more systematic approach? Maybe we can auto-disconnect from anyone who connects to us or sends us a message and who we don't recognize?